distributed/protocol/serialize.py

```diff
@@ -178,6 +220,36 @@ def deserialize(header, frames, deserializers=None):
     --------
     serialize
     """
+    if "is-collection" in header:
+        headers = header["sub-headers"]
+        lengths = header["frame-lengths"]
+        cls = pickle.loads(header["type-serialized"])
+
+        start = 0
+        if cls is dict:
+            d = {}
+            for h, l in zip(headers, lengths):
+                k = h.pop("key")
+                try:
+                    k = msgpack.loads(k, raw=False)
+                except Exception:
+                    k = deserialize(*h.pop("key"), deserializers=deserializers)
```

Comment:
We can drop this code if we you are ok with https://github.com/dask/distributed/pull/2948/files#r314721358

Reply:
Done.